### ISSUE TITLE
Design the article page for desktop

### ISSUE THREAD

**Issue by gissoo** [Princeton-CDH/startwords#57]

**Comment by gissoo** [comment]
@gwijthoff @thatbudakguy @rlskoeser Please take a look at my proposed versions of design resolutions for this issue: (https://www.figma.com/file/MEt3gfKX4N3WTLPbLGBbGC/Startwords?node-id=384%3A2382)

Note: Design versions are put together in clusters. Please look at one cluster at a time (please avoid jumping in between clusters till you're done with one), please read my notes and the label/name of each canvas carefully as nothing will make sense without reading them. – If they still don't make sense after you read them it will be better to meet so I can go through all of them at once.

**Comment by rlskoeser** [comment]
In v1, I like the wider image grid; I do not like the full weaving. Slight preference for tablet grid over the wider grid.

On v2, I like the marginal notes on white (I think - although I wonder if it calls attention to them a little more?). The wider marginal notes are easier to read but the main text looks too cramped to me. I like the second version within v2 best.

These designs haven't been updated to show the actual DOI yet, right? I had a question about how you're showing it elsewhere: I'm not sure how how recognizable it is without the label "DOI", and I wanted to know what @gwijthoff thought about that.

**Comment by rlskoeser** [comment]
@gissoo I think I missed the last section when I commented before! On marginal notes interactions. 

I prefer v1. I don't like the idea of the body text shifting when you select a note (v3/v2); I find that disorienting. The main text also looks pretty cramped in v2 and v3 with the notes selected.

If I understand correctly, the note will display when you click on the footnote. When does a note stop displaying? 

I asked a similar question about the PDF too, but is there any way to give a visual nod to the contextual notes on mobile/tablet in the way that notes are displayed on desktop?

**Comment by gissoo** [comment]
@rlskoeser thanks for writing. I agree with your preferences. 

Thanks for noting about the DOI. I had it updated everywhere else except here. I just updated it. I have added what you're saying only to the screens on this page – with the "DOI: link URL" – but my question is if someone is familiar with the concept of DOI wouldn't they know what this link is? (Therotically it is nice to always be inclusive in the language, however I'm wondering how adding "DOI:" will help someone understand the concept if they are not familiar with it). I am interested in hearing @gwijthoff's thoughts too. 

Yes, the note will display when you click on the footnote, thanks for the question. I forgot to display that part – I think there are so many questions I had before I get there. (I was thinking to have a "x" by the displayed note, or click on the footnote again to hide the note). 

Regarding your last question, this is worth talking about, I'm not sure I understand what you intend to communicate.

**Comment by thatbudakguy** [comment]
overall i prefer v1 for all of these. agree that it's better not to shift the text if we can. i don't think the marginal notes need a white background, but don't have a strong preference.

the way that the white background behind the icon links slips _under_ the weaving background image is super cool! might be a little hard to replicate, but i really like that.

**Comment by gissoo** [comment]
@rlskoeser @thatbudakguy @gwijthoff sorry, I just added one more version (v4), please check it out and let me know what you think. (Please read the note next to it.)

**Comment by thatbudakguy** [comment]
i think if we were going to do v4 it would need to be more customized to the desktop screen in some way, rather than looking the same as the mobile version. not sure how to do that, though.

thinking more about v1, it seems like a very google docs-like interface (thinking of the "suggest changes" function). that interaction is fairly complicated, though - it manages multiple notes in close proximity to each other by ensuring they don't overlap in the sidebar, but also placing them as close to their context as possible. i'm not sure offhand how to implement it, and it certainly sounds like it would require javascript to do correctly (which it might regardless, and that's fine).

one big difference is that in google docs the marginal notes are always visible; clicking on the text anchor just focuses them/moves them around. in v1, it seems like notes aren't always visible (and maybe only one is visible at a time). if we only see one at a time, that might keep things simpler.

**Comment by gissoo** [comment]
@thatbudakguy I agree that it will need to be more customized for desktop, however to me it won't require drastic changes, and it will mostly work as is. 

I am leaning towards the notes not be visible from the beginning, it's one thing if they are always visible (meaning as one enters the page), and it's another thing if the user clicks on one and it will remain visible till the user chooses to hide it. (I think as I mentioned earlier in issue 43, having the notes be visible at all time defeats the purpose them being marginal, to me it's not logical at all, because it's not only about where they appear but also when they appear, they are secondary to the main body, so it's better if they are not always there till the user chooses to see them)

**Comment by gwijthoff** [comment]
My main takeaway: I agree with your last comment, @gissoo , that notes shouldn’t be visible from the beginning.

First variations: I prefer v2 for its shorter line length / wider margins, and I like the bigger images (which I think is the defining feature of v2 over v1, though I could be missing other differences between the two).

Second variations: I prefer v1, and like how it maintains some of the left margin for the main text. The white background for footnotes makes the page feel a bit too chunky and cuts things up in a way that’s not intuitive. There is a lot of text on the page at any one time, however!

Which is why I *love* the next two sets of variations showing interactions leading to viewing contextual notes. Of them, I prefer your fourth set of variations (the final two screens) with the purple background and irregular angles for the note. I agree that an “X” to close the contextual note would be helpful. 

Re: DOI, I think DOI: [link] is sufficient. It’s widely used in academic citation practice and I don’t believe it needs an explainer (at least on our site). It’s just a google away if someone needs it.

**Comment by gwijthoff** [comment]
Just one question @gissoo : I'm still not sure what the text "We call the contextual notes, rather than footnotes!" is standing in for in that final set of variations. Or is that text intended to actually display on the website when a user opens a note?

**Comment by gissoo** [comment]
@gwijthoff Thanks for writing. Grant regarding the text inside the purple container, I was originally proposing that to be there as is, but not anymore, I find it somewhat unprofessional and out of place for this domain :D ...Please let me know if you have any thoughts about it. I meant to delete it in this version, but I forgot, too many pieces!!

**Comment by rlskoeser** [comment]
Documenting my thoughts per our conversation earlier today: I like the idea of some version of the notes in the purple container for multiple reasons.
- Closer to the styles and functionality for mobile and template, which I think is important since both @gwijthoff and @gissoo said that contextual notes is an MVP feature that will set this publication apart
- More distinct/unique; footnotes on the side seems like a fairly familiar solution, and might also cause problems with spacing and interactions (pushing content down the page, multiple notes at once); I also don't like giving up so much of the page content space up to make room for notes.
- Simpler user interaction if we only allow viewing one note with the text at a time
- I think/hope we can use the same implementation across screens (although need to know more about how it will work to be sure.)

**Comment by thatbudakguy** [comment]
Documenting my thoughts as well - I agree with @rlskoeser that we should move forward with the "popup" style version on desktop.
- it's nice to have a similar UX across platforms and I agree that this experience is most similar to mobile/tablet; hopefully it will keep things simple
- definitely agree that notes on the side are a familiar solution, and that having to reserve (or dynamically create) space for them is a challenge

Right now my main reservation is that it might be complicated to implement in an accessible way (and I get the feeling that that way will involve javascript), but I think we can take those problems as they come and hopefully come up with a workable implementation.

**Comment by gissoo** [comment]
@rlskoeser @gwijthoff @thatbudakguy I have revised the designs and interactions. 
Please let me know what you think (Please look under "NEW DESIGN" on the same page, shared at the top of this issue, please also read the notes I have written next to the designs)

**Comment by gwijthoff** [comment]
This looks fantastic! Really love the shadow effect, and the ghost text in the background. Great work, @gissoo

**Comment by thatbudakguy** [comment]
nice work! I like v4a since it keep the note with the text. speaking of which - does the note scroll with the text when the user scrolls the page (on tablet and desktop, at least)? otherwise it would float around and be next to different text after scrolling. i feel like we discussed this before, but i can't remember what the answer was....sorry!

**Comment by gissoo** [comment]
@gwijthoff @thatbudakguy thanks for writing!

@thatbudakguy I've written about this a bit in #4 note next to the design (which includes the highest preference for now – and what RSK and I agreed might be easier as well) – I don't think it should still be visible once the scroll is "activated" – we were thinking it would be visible and scroll with the text till the text with the superscript is no longer visible then the container will also disappear but that is not a good idea (because the user in many cases may want to read the line below which is being covered by the container) – better if it disappears once there is any scroll activity if the user doesn't close the container on their own

**Comment by thatbudakguy** [comment]
ah - I didn't put two and two together about the scrolling behavior; obviously if the note disappears on scroll then it doesn't matter if it's fixed or not. again, thanks for explaining.

**Comment by rlskoeser** [comment]
I like these designs a lot. The thing I'm concerned about is the "text associated with the hyperlink" - there's no association in the markup and I'm not sure how we could infer one automatically. The designs indicate that the words before the footnote are clickable as if they are part of the footnote reference, but that's now how our footnotes work. I can't think of a way to implement that - @thatbudakguy can you?

Similarly, the design has the footnote arrow pointing to the words before the reference; I think it needs to be pointing to the reference number.

**Comment by gissoo** [comment]
@rlskoeser I see what you are saying. The problem is that the superscript is such a small point for selection. I agree that the arrow could be pointing to the reference number rather than words (but selection is still an issue)

**Comment by rlskoeser** [comment]
@gissoo good point. I have a thought about that, let me try something

**Comment by gissoo** [comment]
@rlskoeser I did change where the arrow is pointed

**Comment by gissoo** [comment]
@rlskoeser I think since we discussed on the potential issues with selection you have addressed it by making at least the two words before the superscript + the superscript clickable (Thank you so much!! This helps a lot)

@gwijthoff @rlskoeser @thatbudakguy How can we close this issue? Do you see/find anything in the designs that doesn't make sense or points to unanswered questions? (I have archived all the screens attached linked to this issue that are now irrelevant, so you won't see them here anymore)

**Comment by rlskoeser** [comment]
@gissoo could you summarize where things stand? I've kind of lost track.

The one concern I have is that the clickable space before the superscript is currently an arbitrary width, we don't have a way to automatically highlight the two words preceding.

**Comment by gissoo** [comment]
@rlskoeser it will be most helpful to only look at and mainly read what is in this [figma file here](https://www.figma.com/file/MEt3gfKX4N3WTLPbLGBbGC/Startwords?node-id=384%3A2382) – I made sure the notes are updated. This issue is mainly focused on the behaviors associated with the contextual note  

- I did add the tablet and mobile on this page in case it's useful to see them here (even though this issue was specifically made for desktop) – if you think it's confusing I can remove them. 

- re clickable space: we don't have a way to define a fixed width? I think as long as it covers a bit more than the superscript to include a bit of the area around it, it's much better than nothing.
